File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        
        # result = nums[0]
        # left, right = 0, len(nums) - 1

        # while left < right: 
        #     if nums[left] < nums[right]: 
        #         return min(result, nums[left]) 

        #     mid = left + (right - left) // 2
        #     result = min(nums[mid], result)
        #     if nums[mid] >= nums[left]: 
        #         left = mid + 1
        #     else: 
        #         right = mid - 1
        # return result 

        return min(nums) 

File: Data Structures & Algorithms/binary-search/submission-1.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int left = 0; // left index
        int right = nums.size() - 1; // right index  

        while(left <= right) { // condition to end the while loop 
            int middle = left + ((right - left) / 2); 
            if(nums[middle] == target) { 
                return middle; 
            } 

            if(nums[middle] < target) { 
                left = middle + 1; 
            } else {
                right = middle - 1;
            }

        }
        return -1; 
    }
};
